components/public-api-server/pkg/identityprovider/cache_test.go

```diff
@@ -111,7 +111,7 @@ func TestRedisCachePublicKeys(t *testing.T) {
 		t.Run(test.Name, func(t *testing.T) {
 			s := miniredis.RunT(t)
 			client := redis.NewClient(&redis.Options{Addr: s.Addr()})
-			cache := NewRedisCache(client)
+			cache := NewRedisCache(context.Background(), client)
```

Comment:
You'll want to cancel the context, once the test is done else you're leaking the sync routine in tests, which then also likely affects subsequent tests